src/conventional/metar_csv2ioda.py

```diff
@@ -152,99 +162,86 @@ def _rd_metars(self):
-        self.outdata[('station_elevation', 'MetaData')] = np.array(data['ob_elev'], dtype=np.float32)
-        self.outdata[('height', 'MetaData')] = np.array(data['ob_hght'], dtype=np.float32)
-        iodavar = 'surface_pressure'
-        self.outdata[(iodavar, iconv.OvalName())] = np.array(data['ob_psfc'], dtype=np.float32)
-        self.outdata[(iodavar, iconv.OerrName())] = np.full((nlocs), 200.0, dtype=np.float32)
-        self.outdata[(iodavar, iconv.OqcName())] = np.full((nlocs), 2, dtype=np.int32)
-        iodavar = 'air_temperature'
-        self.outdata[(iodavar, iconv.OvalName())] = np.array(data['ob_temp'], dtype=np.float32)
-        self.outdata[(iodavar, iconv.OerrName())] = np.full((nlocs), 0.2, dtype=np.float32)
-        self.outdata[(iodavar, iconv.OqcName())] = np.full((nlocs), 2, dtype=np.int32)
-        iodavar = 'specific_humidity'
-        self.outdata[(iodavar, iconv.OvalName())] = np.array(data['ob_spfh'], dtype=np.float32)
-        self.outdata[(iodavar, iconv.OerrName())] = np.full((nlocs), 0.75E-3, dtype=np.float32)
-        self.outdata[(iodavar, iconv.OqcName())] = np.full((nlocs), 2, dtype=np.int32)
-        iodavar = 'eastward_wind'
-        self.outdata[(iodavar, iconv.OvalName())] = np.array(data['ob_uwnd'], dtype=np.float32)
-        self.outdata[(iodavar, iconv.OerrName())] = np.full((nlocs), 0.7, dtype=np.float32)
-        self.outdata[(iodavar, iconv.OqcName())] = np.full((nlocs), 2, dtype=np.int32)
-        iodavar = 'northward_wind'
-        self.outdata[(iodavar, iconv.OvalName())] = np.array(data['ob_vwnd'], dtype=np.float32)
-        self.outdata[(iodavar, iconv.OerrName())] = np.full((nlocs), 0.7, dtype=np.float32)
-        self.outdata[(iodavar, iconv.OqcName())] = np.full((nlocs), 2, dtype=np.int32)
-
-        DimDict['nlocs'] = nlocs
-        AttrData['nlocs'] = np.int32(DimDict['nlocs'])
+        nlocs = len(data['dateTime'])
+        DimDict['Location'] = nlocs
+        AttrData['Location'] = np.int64(DimDict['Location'])
+
+        # Set units of the MetaData variables and all _FillValues.
+        for key in meta_keys:
+            dtypestr = locationKeyList[meta_keys.index(key)][1]
+            self.varAttrs[(key, metaDataName)]['_FillValue'] = missing_vals[dtypestr]
+            this_units = locationKeyList[meta_keys.index(key)][2]
+            if this_units:
+                self.varAttrs[(key, metaDataName)]['units'] = this_units
+            self.outdata[(key, metaDataName)] = np.array(data[key], dtype=dtypes[dtypestr])
+
+        # Transfer from the 1-D data vectors and ensure output data (obs_data) types using numpy.
+        for n, iodavar in enumerate(obsvars):
+            self.outdata[(iodavar, obsValName)] = np.array(data[iodavar], dtype=np.float32)
+            self.outdata[(iodavar, obsErrName)] = np.full(nlocs, obserrlist[n], dtype=np.float32)
+            self.outdata[(iodavar, qcName)] = np.full(nlocs, 2, dtype=np.int32)
```

Comment:
Thanks Cory.  Basically all observations are given preQC=2 value because that is what EMC uses for a default value for variables that have not been QC-checked at all yet.  Clearly this starting value would be clobbered by UFO processing so this is just to indicate no QC done yet following what ObsProc specifies for 0-15 range.

Reply:
@gthompsnJCSDA Could you add a comment in the converter indicating the meaning of this value. When the ioda enumerated type PR is merged (well after this sprint), then we will want to go through and add comments to the files.

Reply:
Addressed Ryan's comment so resolving the conversation.